app/Platform/Services/GameSuggestions/Strategies/SimilarGameStrategy.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services\GameSuggestions\Strategies;
+
+use App\Models\Game;
+use App\Platform\Data\GameData;
+use App\Platform\Data\GameSuggestionContextData;
+use App\Platform\Enums\GameSetType;
+use App\Platform\Enums\GameSuggestionReason;
+
+class SimilarGameStrategy implements GameSuggestionStrategy
+{
+    private ?Game $selectedGame = null;
+
+    public function __construct(
+        private readonly Game $sourceGame,
+    ) {
+    }
+
+    public function select(): ?Game
+    {
+        // Get game set IDs for similar games that contain our source game.
+        $gameSetIds = $this->sourceGame->gameSets()
+            ->whereType(GameSetType::SimilarGames)
+            ->pluck('game_sets.id');
+
+        if ($gameSetIds->isEmpty()) {
+            return null;
+        }
+
+        $this->selectedGame = Game::query()
+            ->whereHas('gameSets', function ($query) use ($gameSetIds) {
+                $query->whereIn('game_sets.id', $gameSetIds);
+            })
+            ->where('ID', '!=', $this->sourceGame->id)
+            ->whereHasPublishedAchievements()
+            ->inRandomOrder()
+            ->first();
```

Comment:
Not sure how these are similar
![image](https://github.com/user-attachments/assets/31c93be9-da7f-484e-a229-2f39a1367ef5)

![image](https://github.com/user-attachments/assets/f638601a-e28d-4b1d-a8cd-d608bd508920)

It is an interesting choice though, as I could see them being related, but the UI doesn't reflect that.

They do both share a common related game (Boxxle).
Carrier (17588) and Pile Wonder (10793) were both also suggested as a similar game and are linked to Boxxle.